assets/js/modules/search-console/components/widgets/PopularKeywordsWidget.js

```diff
@@ -21,31 +21,120 @@
  */
 import PropTypes from 'prop-types';
 
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+
 /**
  * Internal dependencies
  */
 import Data from 'googlesitekit-data';
 import { CORE_USER } from '../../../../googlesitekit/datastore/user/constants';
+import {
+	DATE_RANGE_OFFSET,
+	MODULES_SEARCH_CONSOLE,
+} from '../../datastore/constants';
+import { generateDateRangeArgs } from '../../util';
+import { numFmt } from '../../../../util';
+import Link from '../../../../components/Link';
+import MetricTileTable from '../../../../components/KeyMetrics/MetricTileTable';
+import useViewOnly from '../../../../hooks/useViewOnly';
+import { whenKeyMetricsWidgetVisible } from '../../../../components/KeyMetrics';
+import { ZeroDataMessage } from '../common';
+
+const { useSelect, useInViewSelect } = Data;
+
+function PopularKeywordsWidget( { Widget } ) {
+	const viewOnlyDashboard = useViewOnly();
+
+	const dates = useSelect( ( select ) =>
+		select( CORE_USER ).getDateRangeDates( {
+			offsetDays: DATE_RANGE_OFFSET,
+			compare: true,
+		} )
+	);
 
-const { useSelect } = Data;
+	const reportOptions = {
+		...dates,
+		dimensions: 'query',
+		limit: 100,
+	};
 
-export default function PopularKeywordsWidget( { Widget, WidgetNull } ) {
-	const keyMetricsWidgetHidden = useSelect( ( select ) =>
-		select( CORE_USER ).isKeyMetricsWidgetHidden()
+	const report = useInViewSelect( ( select ) =>
+		select( MODULES_SEARCH_CONSOLE ).getReport( reportOptions )
 	);
 
-	if ( keyMetricsWidgetHidden !== false ) {
-		return <WidgetNull />;
-	}
+	const loading = useInViewSelect(
+		( select ) =>
+			! select( MODULES_SEARCH_CONSOLE ).hasFinishedResolution(
+				'getReport',
+				[ reportOptions ]
+			)
+	);
+
+	const keywordsDateRangeArgs = generateDateRangeArgs( dates );
+
+	const columns = [
+		{
+			field: 'keys.0',
+			Component: ( { fieldValue } ) => {
+				const searchAnalyticsURL = useSelect( ( select ) => {
+					return ! viewOnlyDashboard
+						? select( MODULES_SEARCH_CONSOLE ).getServiceReportURL(
+								{
+									...keywordsDateRangeArgs,
+									query: `!${ fieldValue }`,
```

Comment:
I find this `!` prefix a bit obscure, would you mind adding an inline comment to clarify it?

Reply:
Thanks, sorry to be a bit pedantic though but the comment doesn't really clear things up all that much as it doesn't explain _how_ the SC console uses the query parameter, simply that it _does_ use it which is quite easy to figure out.

I've taken a look and it's evident that the SC uses the `!` prefix to indicate an exact match for the query, rather than say a substring match, so I'd suggest updating the comment accordingly, something like this:

> The exclamation mark at the beginning of the query specifies that the term should be treated as an exact match on the Search Console search results page.

Screengrab of the SC "New query" UI for reference:

![image](https://github.com/google/site-kit-wp/assets/18395600/21567f4a-6fe5-4624-bb32-7a515fc1a104)